examples/bioasq9b.py

```diff
@@ -0,0 +1,232 @@
+                    "context": datasets.Value("string"),
+                    "answer": datasets.Sequence(datasets.Value("string")),
+                }
+            )
+
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(list(_URLs.values()))
+        # BioASQ test data is split into multiple records {9B1_golden.json,...,9B5_golden.json}
+        # combine these files into a single test set 9Bx_golden.json
+        if not os.path.exists(
+            os.path.join(data_dir, "Task9BGoldenEnriched/9Bx_golden.json")
+        ):
+            pass
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": os.path.join(
+                        data_dir, "BioASQ-training9b/training9b.json"
+                    ),
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": os.path.join(
+                        data_dir, "BioASQ-training9b/training9b.json"
+                    ),
+                    "split": "test",
+                },
+            ),
+        ]
+
+    def _generate_examples(
+        self,
+        filepath,
+        split,  # method parameters are unpacked from `gen_kwargs` as given in `_split_generators`
+    ):
+        """Yields examples as (key, example) tuples."""
+        # This method handles input defined in _split_generators to yield (key, example) tuples from the dataset.
+        # The `key` is here for legacy reason (tfds) and is not important in itself.
+
+        with open(filepath, encoding="utf-8") as file:
+            data = json.load(file)
+            for record in enumerate(data):
+                if self.config.name == "original":
+                    yield record["id"], {
```

Comment:
I might be confused, but I think `record` here will be a tuple with the first element i = 0 to len(data)-1 and the second element being data[i]  .. so `record["id"]` seems like it would throw an error. I think maybe we want something like, 
```
for _id, record in enumerate(data):
    yield _id, {"id": record["id"], ...}
```